web/api/v1/api.go

```diff
@@ -375,6 +383,30 @@ func (api *API) targets(r *http.Request) (interface{}, *apiError) {
 	return res, nil
 }
 
+type AlertmanagerTarget struct {
+	// Labels before any processing.
+	DiscoveredLabels model.LabelSet `json:"discoveredLabels"`
+	// Any labels that are added to this target and its metrics.
+	Labels model.LabelSet `json:"labels"`
+
+	URL string `json:"url"`
+}
+
+func (api *API) alertmanagers(r *http.Request) (interface{}, *apiError) {
+	amTargets := api.alertmanagerRetriever.AlertmanagerTargets()
+	res := make([]*AlertmanagerTarget, len(amTargets))
+
+	for i, t := range amTargets {
+		res[i] = &AlertmanagerTarget{
+			DiscoveredLabels: t.DiscoveredLabels(),
```

Comment:
We don't use labels like we do for normal targets, so I don't think we should expose them to users to avoid confusion.